base/abstractarray.jl

```diff
@@ -1835,8 +1835,8 @@ end
 
 ## N argument
 
-ith_all(i, ::Tuple{}) = ()
-ith_all(i, as) = (as[1][i], ith_all(i, tail(as))...)
+@inline ith_all(i, ::Tuple{}) = ()
```

Comment:
Probably the problem is only with the one below that calls itself recursively, but I don't think it could harm to force inlining on the two.